docs/src/pages/components/data-grid/selection/selection.md

```diff
@@ -62,8 +62,16 @@ In the demo below only rows with quantity above 50000 can be selected:
 
 ## Controlled selection
 
+Use the `selectionModel` prop to control the grid selection. Each time the `selectionModel` changes the `onSelectionModelChange` callback is called with the new selection model value.
+
 {{"demo": "pages/components/data-grid/selection/ControlledSelectionGrid.js", "bg": "inline"}}
 
+### Controlled selection and server-side pagination
+
+Using controlled selection when `paginationMode="server"` can result in the `DataGrid` clearing the provided `selectionModel` on page change. Depending on your implementation of the server-side pagination, when the page changes and there are no rows in the grid with ids equal to the `GridRowId`s provided to the `selectionModel` the grid will clear the `selectionModel`. To prevent this either save the `selectionModel` and restore it later or append the newly loaded grid rows to the existing rows.
```

Comment:
> and restore it later

Some people will probably struggle to implement it.
Maybe we could also add an example for this part, or just take for granted that appending the rows is the way we want to encourage.

Reply:
> Maybe we could also add an example for this part

This would be greatly appreciated if you can accommodate.

Reply:
I'm opened to suggestions here. It's unclear what the desired behaviour should be. For me, the way it is looks correct but so does the initial implementation in the ticket (not appending the rows). Because the selection is controlled the only solution I can see is either to leave it as it is or prevent the grid from cleaning up the `selectionModel` if the rows are missing.

Reply:
We should probably also take this https://github.com/mui-org/material-ui-x/pull/2602#issuecomment-919173644 into consideration. If we want to fix that as well we will need to change the way the `selectionModel` behaves as the Select All feature is not really server-side friendly at the moment.

Reply:
Indeed